eng/deploy.yaml

```diff
@@ -33,7 +77,7 @@ stages:
       displayName: Notify dotnet-eng-staus/start
       continueOnError: true
       inputs:
-        serviceConnection: ${{ parameters.DotNetStatusEndpoint }}
+        serviceConnection: $(DotNetStatusEndpoint)
```

Comment:
This won't work. Service connections can't be variables.  You'll need to make this a template and include it from the actual pipeline so that you can do template parameters (like it used to be).

This build is so short that this seems like a big, complicated change for little gain.

Reply:
Essentially, this file needs to have no changes at all, it just needs to be included from a new pipeline.  Though there might be a single change to remove the dependency on the build phase.

Reply:
The variables were populated in my test run - https://dev.azure.com/dnceng/internal/_build/results?buildId=483603&view=results

Reply:
They didn't deploy, however. Unless something drastic has changed since a few weeks ago, this will not work.

Reply:
You'll note the "report build start" didn't have the secret variable, because it returned HTML.  Because it failed to load the variable group containing the secret, so it didn't pass the authentication cookie.  It shouldn't have redirected, but I expect that's because some random thing is running in staging at the moment.

Reply:
Actually, the URL appears to just be messed up, even though I fixed that recently... wonder how it's wrong again.

Reply:
Actually, no, it's fine... it just redirected, which seems to imply my recent changes aren't running (or maybe not correct).  Anyway, it still doesn't look like the secrets got passed.

Reply:
Moved to templates